### PR TITLE
GitHub actions integration

### LINKED ISSUE #163: Move to github actions and pure Node.js linting
**Problem description**
Currently there are a couple of dependencies to dotnet and travis that can be replaced with github actions ever since it has been introduced. This would make things more accessible to web-developers by leveraging node.js instead which is already in use anyhow.

**Desired Solution**
* Removal of Jenkins pipeline
* Creating appropriate github actions
* Rewrite tests with node.js

### PR DESCRIPTION
**Pull request by jensstalder**
## Semantic versioning

This PR is a **Patch** (bugfix, hotfix, without adding new stuff) 

## Generall things to consider 
- I have read the [**CONTRIBUTING.md**](https://github.com/qualipool/swissrets/blob/master/CONTRIBUTING.md) document.
- If my change requires documentation, I added it already.
- I have added tests to cover all my changes.
- All new and existing tests pass.

## List of issues related to this PR
- closes #163

Problem description
Currently there are a couple of dependencies to dotnet and travis that can be replaced with github actions ever since it has been introduced. This would make things more accessible to web-developers by leveraging node.js instead which is already in use anyhow.

Desired Solution

Removal of Jenkins pipeline
Creating appropriate github actions
Rewrite tests with node.js

### CODE REVIEW

**Comment by jensstalder** [comment]
Looks like some of the docs/docs/html changes have made it into this branch. I/we will need to update theses back to the master representation, since these files should not change based on a PR. (only when release version tag gets set).

File: package.json [hunk 1]
```diff
@@ -32,16 +33,19 @@
...
+    "standard": "^16.0.3"
   },
   "nodemonConfig": {
     "watch": [
       "examples/*",
       "schema/schema.xsd"
     ]
+  },
+  "dependencies": {
+    "xmllint": "^0.1.1"
```
Comment (jensstalder) [inline package.json:49]: Will probably need to be moved to devDependencies

**Comment by jensstalder** [comment]
Just noticed that there is a docker-compose.yml file that sets-up the dotnet testing within a docker container. Intresting.

File: scripts/xmllint/index.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
+const xmllint = require('xmllint')
+const fs = require('fs')
+const path = require('path')
+const schemaPath = path.join(__dirname, '..', '..', 'schema', 'schema.xsd')
+const glob = require('glob')
+const { log } = require('../lib')
+// process.setMaxListeners(100);
```
Comment (koffeinfrei) [inline scripts/xmllint/index.js:7]: Remove this line (or uncomment)?

File: scripts/xmllint/index.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        if (err) {
+          log.failure(err)
+        } else {
+          const result = xmllint.validateXML({
+            xml,
+            schema
+          })
+
+          if (!result.errors) {
+            log.failure(`❌ file should have failed but it did not: ${file}`)
```
Comment (koffeinfrei) [inline scripts/xmllint/index.js:55]: Minor: **F**ile

File: scripts/xmllint/index.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+            process.exit(1)
+          } else {
+            log.success(`✅ errors found and expected for this file: ${file}`)
```
Comment (koffeinfrei) [inline scripts/xmllint/index.js:58]: Minor (for consistency with the other messages): **E**rrors

File: package.json [hunk 5]
```diff
@@ -7,11 +7,12 @@
...
     "lint": "standard",
+    "lint:fix": "standard --fix",
+    "xmllint": "node scripts/xmllint",
     "xsddoc": "node scripts/xsddoc",
     "start": "nodemon --exec 'npm test -s'",
-    "test": "dotnet test scripts/tests/SwissRETS.Tests.csproj",
-    "postversion": "node scripts/postversion",
-    "xsddocs": "node scripts/xsddocs"
+    "test": "node scripts/xmllint",
+    "postversion": "node scripts/postversion"
```
Comment (koffeinfrei) [inline package.json:15]: Potentially rename to `publish-version`. "Post" confused me a little, as it sounds rather like a post hook.

**Comment by koffeinfrei** [comment]
* `package-lock.json` is outdated.
* Should we add an `.nvmrc` (or any other alternative you prefer) to lock the node version?

Otherwise LGTM :+1:

**Pull request merged**